packages/components/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import map from "lodash/map";
+import isNaN from "lodash/isNaN";
+import { Box } from "../../primitives/Box";
+import { Button } from "../Button";
+import { PaginationButton } from "./PaginationButton";
+import { usePagination } from "./usePagination";
+
+export interface PaginationProps
+  extends Omit<React.HtmlHTMLAttributes<HTMLDivElement>, "color"> {
+  /**
+   * The total number of pages.
+   */
+  totalPages: number;
+  /**
+   * The currently selected page.
+   */
+  currentPage: number;
+  /**
+   * Callback executed when the page is changed.
+   */
+  onPageChange: (page: number) => void;
+  /** The accessible label assigned to the pagination. */
+  label?: string;
+  /**
+   * The number of pages to display to each side of the current page.
+   */
+  pagesAroundCurrentPage?: number;
+}
+
+/** Pagination lets users navigate through content or a dataset that’s been broken up into multiple pages. */
+const Pagination = React.forwardRef<HTMLDivElement, PaginationProps>(
+  (
+    {
+      currentPage,
+      label = "Page navigation",
+      onPageChange,
+      totalPages,
+      pagesAroundCurrentPage = 1,
+      ...props
+    },
+    ref
+  ) => {
+    const pages = usePagination(
+      totalPages,
+      pagesAroundCurrentPage,
+      currentPage
+    );
+
+    const nextPage = () => {
+      if (currentPage !== totalPages) onPageChange(currentPage + 1);
+    };
+    const prevPage = () => {
+      if (currentPage >= 1) onPageChange(currentPage - 1);
+    };
+
+    return (
+      <Box.nav aria-label={label} ref={ref} role="navigation" {...props}>
+        <Box.ul
+          alignItems="center"
+          display="flex"
+          gap="space30"
+          listStyleType="none"
+          margin="space0"
+          padding="space0"
+        >
+          <Button
+            aria-label="Previous page"
+            disabled={currentPage === 1}
+            iconOnly
+            leadingIcon="ChevronLeftIcon"
+            onClick={prevPage}
+            style={{ outlineColor: "colorBorderPrimary" }}
+            variant="ghost"
+          />
+          {map(pages, (page, index) => {
+            if (page === "...") {
+              return (
+                <Box.div
+                  display="inline-flex"
+                  fontFamily="fontFamilyModerat"
+                  fontSize="fontSize20"
+                  fontWeight="fontWeightMedium"
+                  justifyContent="center"
+                  key={index}
+                  paddingBottom="space30"
+                  paddingLeft="space30"
+                  paddingRight="space30"
+                  paddingTop="space30"
+                  w="36px"
+                >
+                  ...
+                </Box.div>
+              );
+            }
+            return (
+              <PaginationButton
+                isCurrentPage={currentPage === Number(page)}
+                key={index}
```

Comment:
We shouldn't be using [indexes as keys](https://react.dev/learn/rendering-lists#why-does-react-need-keys). The page might be a better index.